### ISSUE TITLE
Terminal App Version 1.8.1521.0 crashes on openssh client connection exit (WSL2 Ubuntu buster/sid Linux)

### ISSUE THREAD

**Issue by ministry-of-code** [microsoft/terminal#10657]
### Windows Terminal version (or Windows build number)

Terminal App Version 1.8.1521.0

### Other Software

openssh on Linux WSL2 (Ubuntu)

### Steps to reproduce

Sometimes, not always, the Terminal App crashes when I log off of a SSH connection to a remote (Linux) Server. I run WSL2 with Ubuntu Linux 4.19.84-microsoft-standard+ #1 SMP Fri Mar 19 11:33:09 CET 2021 x86_64 x86_64 x86_64 GNU/Linux.

This is really really a showstopper for my productive work. I have many tabs and server connections open and when the terminal app crashes, all those connections and open files, etc are lost!

### Expected Behavior

Terminal app does not crash

### Actual Behavior

Terminal app crashes

**Comment by zadjii-msft** [comment]
Could you repro this with the Feedback hub? That'll get us a stack trace we can use to root-cause this

/feedback

**Comment by ministry-of-code** [comment]
> Could you repro this with the Feedback hub? That'll get us a stack trace we can use to root-cause this
> 
> /feedback

Thanks for the feedback. How would this work? When the app crashes I open the feedback hub app and create an issue? Will the stack trace automatically be attached to the issue? (Never done this before that's why I am asking).

**Comment by ministry-of-code** [comment]
> Could you repro this with the Feedback hub? That'll get us a stack trace we can use to root-cause this
> 
> /feedback

@zadjii-msft thanks for the feedback. I have submitted an issue via the Feedback hub two weeks ago:

https://aka.ms/AAd71do

Does this help?

**Comment by zadjii-msft** [comment]
Oh no, just open the feedback hub and draft up the feedback, then the last step will ask you for a recording of the issue. Start the recording, then switch over to the Terminal and repro the bug, then switch back and end the recording. The Feedback Hub will automatically collect up the crash that happened during the recording and link it with the feedback. Submit the feedback, and share the aka.ms URL here, and we can then go look up the crash on the backend. Neat!

**Comment by zadjii-msft** [comment]
(unfortunately, the original feedback you filed didn't collect up the crash, so it doesn't have anything helpful in it 😢)

**Comment by ministry-of-code** [comment]
Ok, I understand. Unfortunately, I can not start the diagnose process in the Feedback hub app. When I try to start it for the windows-terminal app, I get an error message telling me that it did not work and I should try it again:

![image](https://user-images.githubusercontent.com/2571417/125629526-052f9e0c-8dc7-4b8a-9839-47565e1ca1d6.png)

**Comment by WSLUser** [comment]
To ensure WSL2 isn't involved, are you using the default kernel or a custom one? If it's custom, what's different from the default WSL2 kernel? Technically this shouldn't matter unless something with the console options was changed or possibly even network config since ssh is involved.

**Comment by ministry-of-code** [comment]
> To ensure WSL2 isn't involved, are you using the default kernel or a custom one? If it's custom, what's different from the default WSL2 kernel? Technically this shouldn't matter unless something with the console options was changed or possibly even network config since ssh is involved.

You're right, I am using a custom WSL2 Kernel Configuration (based on the official MS WSL2-Kernel). I can't remember why I did this (it's a couple of months ago that I compiled it) - but it must have had an important reason - probably to fix a network related problem (docker, localhost forwarding or such). I will disable the custom kernel and use the official one again and see if this solves the crash or not and will come back in a few days to report.
